src/univariate/continuous/triangular.jl

```diff
@@ -93,20 +93,34 @@ entropy(d::TriangularDist{T}) where {T<:Real} = one(T)/2 + log((d.b - d.a) / 2)
 
 function pdf(d::TriangularDist, x::Real)
     a, b, c = params(d)
-    return if x < c
-        2 * max(x - a, 0) / ((b - a) * (c - a))
+    t = promote(x, a)[1]
```

Comment:
The problem with heuristics in general is that it is easy to miss some cases. Therefore usually in Distributions the typical approach is to perform computations as much as possible (possibly by clamping values to the support) and handle special cases (such as out of support) only afterwards, based on the type of these computations. In that way we can avoid having to implement any heuristics.

One example where the current heuristic fails:
```julia
julia> @inferred pdf(TriangularDist{Int}(0, 2, 1), 1)
ERROR: return type Int64 does not match inferred return type Union{Float64, Int64}
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] top-level scope
   @ REPL[5]:1
```

Reply:
Ok, I've given it another shot.  I went back to the form of the functions that were previously there and modified the cdf to accurately handle the case when b=c, the Inf test, and the NaN test.  I tried to follow the "spirit" you've mentioned above.  All tests passed on my local.  I have my fingers crossed.